### ISSUE TITLE
Destination's are skipped after recovering destination messages until channel is restarted

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3339]
RecoveryTask doesn't add previous destinations to enabledMetaDataIds, then it overwrites the enabledMetaDataIds on the chain itself. Then channel starts. Then bad things happen.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3439
Reporter: wayneh
Created: 2014-09-08T20:44:40.000-0700

**Comment by rbeckman-nextgen** [comment]
This should actually no longer happen because of MIRTH-3387, since we're always putting destinationSet in the source map for new messages. Therefore the chain will always have its enabledMetaDataIds set during message processing. If we want to be completely prudent though, we could do the same thing we're doing in Channel.process and reset the enabledMetaDataIds on the chain after recovery finishes.

Imported Comment. Original Details:
Author: narupley
Created: 2014-09-09T08:27:59.000-0700

**Comment by rbeckman-nextgen** [comment]
Fixed in revision 7351. Whenever destination messages are recovered, the chains' enabled metadata IDs are now always being reset.

Imported Comment. Original Details:
Author: narupley
Created: 2014-09-09T15:00:08.000-0700

**Comment by rbeckman-nextgen** [comment]
I was able to replicate this issue in 3.0.3 by creating a channel with about 20 destinations. On two of the destinations, I had Thread.sleep() methods and would alternate between dispatching a message and halting a channel at one of the sleeper states. After doing this for a few moments, I got my channel to a state where the first 4 destinations were being skipped.

I exported/imported this channel into 3.1, and attempting to perform similar testing, I alternated between sending messages and halting the channel at various phases as the message travelled through the destinations. No matter how hard I tried, I was unable to reproduce channel skipping.

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-17T15:22:08.000-0700
